coda_v2_python_client/firebase_client_wrapper.py

```diff
@@ -461,7 +461,7 @@ def set_segment_messages_metrics(self, segment_id, messages_metrics):
         """
         self.get_segment_messages_metrics_ref(segment_id).set(messages_metrics.to_firebase_map())
 
-    def compute_segment_coding_progress(self, segment_id, messages=None):
+    def compute_segment_coding_progress(self, segment_id, messages=None, incremental_update=False):
```

Comment:
I'm finding all these flags starting to get a bit confusing, and I don't think the function name is helping either.

What do you think about breaking this into two:
```
def compute_message_metrics(messages: list of Message): MessageMetrics
def set_message_metrics(segment_id: str, metrics: MessageMetrics): void
```
?

Then it's clearer what each function is doing, and all the options currently being controlled by all these flags just become part of the logic at the use-site, so you don't have to consider as many different control paths here at once.

e.g. to recompute:
```
metrics = compute_message_metrics(get_messages(segment_id))
set_message_metrics(metrics)
```
and to add a message:
```
metrics = get_message_metrics(segment_id) + compute_message_metrics([message])
set_message_metrics(metrics)
```

Reply:
Updated, thanks.